packages/cli/src/lib/generators/CodeGenerator.ts

```diff
@@ -0,0 +1,104 @@
+import { BuildConfig } from "../Compiler";
+import { SchemaComposer } from "../SchemaComposer";
+
+import chalk from "chalk";
+import fs, { readFileSync } from "fs";
+import path from "path";
+import Mustache from "mustache";
+import {
+  OutputDirectory,
+  OutputEntry,
+  writeDirectory,
+} from "@web3api/schema-bind";
+import { TypeInfo, parseSchema } from "@web3api/schema-parse";
+
+export class CodeGenerator {
+  private _schema: string | undefined = "";
+  constructor(private _templateFile: string, private _config: BuildConfig) {}
+
+  public async generateCode(): Promise<boolean> {
+    // Make sure that the output dir exists, if not create a new one
+    if (!fs.existsSync(this._config.outputDir)) {
+      fs.mkdirSync(this._config.outputDir);
+    }
+
+    // Compose schema from manifest
+    const schemaComposer = new SchemaComposer(this._config);
+    const manifest = await schemaComposer.loadManifest();
+    const composedSchema = await schemaComposer.composeSchemas(manifest);
+    const typeInfo = parseSchema(composedSchema.combined!);
+    this._schema = composedSchema.combined;
+
+    // Check the template file if it has the proper run() method
+    // eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports, @typescript-eslint/naming-convention
+    const { run } = await require(this._templateFile);
+
+    if (!run) {
+      console.log(
+        chalk.red(
+          "The template file provided is wrong or doesn't have the 'run' method."
+        )
+      );
+      return false;
+    }
+
+    const output: OutputDirectory = {
+      entries: [],
+    };
+
+    await run(output, {
+      typeInfo,
+      generate: (templatePath: string, typeInfo: TypeInfo) =>
```

Comment:
Instead of the second property being `typeInfo`, we want this to be a generic object so they can pass in whatever they want (might only want a single ObjectTypeDefinition for example). So let's set it to `config: unknown`.

Reply:
Then inside the template, do you want to use sth like `{{config.objectTypes}}`?

Reply:
Yes you could do that. The reason for having `config: unknown` is to support use-cases like this:
`web3api.gen.js`
```
const singleType = typeInfo.objectTypes.find(
  (type) => type.name === "MySpecialType"
);

generate("./SpecialTemplate.mustache", singleType);
```
`SpecialTemplate.mustache`
```
{{name}}: {{type}}
```